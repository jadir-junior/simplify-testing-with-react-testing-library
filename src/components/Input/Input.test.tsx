import { render, screen } from "@testing-library/react";

import { Input } from ".";
import React from "react";

describe("<Input />", () => {
  it("Should render", () => {
    const { container } = render(<Input />);

    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render with a label", () => {
    render(<Input label="username" />);

    expect(screen.getByLabelText(/username/i)).toBeInTheDocument();
  });

  it("should render without a label", () => {
    render(<Input />);

    expect(screen.queryByLabelText(/username/i)).not.toBeInTheDocument();
  });
});
