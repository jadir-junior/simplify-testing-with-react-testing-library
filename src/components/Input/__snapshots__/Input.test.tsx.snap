// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  padding: 8px;
  border: 1px solid #ddd;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  class="c0"
>
  <input
    class="c1"
  />
</div>
`;
